### PR TITLE
Changlings / vampires can now roll hijack

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Changeling and vampires can now roll hijack. Currently, this is a 10% chance per antag, same as traitor. I may lower this to 8%, we shall see, this should not be too common.

~~Changelings gain the following boost when they are hijack:
	chem_storage = 100 //25% more
	chem_recharge_rate = 4 //25% more.
	genetic_points = 12 //20% more.~~

~~Vampires get a 33% reduction to cooldown and blood cost on abilities.~~

The above is disabled for the TM, to see how they do.

## Why It's Good For The Game

We have these antags, but it is rare to see one truely reach it's full potential of power, being limited (and reasonably so), by the rules.

However, what if these antags could go all out, just this once. This lets these antags pull out all the stops antag wise, and even go byond to have slightly better stats, which should help with the fact they can not (normally, looking at you evidence) acquire high powered weaponry to shred through crew, or the ever important emag for a quick launch.
Naturally, vampires / changelings can do most things a traitor can do hijack wise, though a vampire will find it hard to use space to their advantage, as well as make sure they have a lot of blood for shuttle, and changelings will find that they will have to manage even their increased chemical storage much better, as unlike adrenals, they can't keep spamming antistun forever, they will run out of antistun much faster than a traitor would run out of adrenals, though they have fleshmend to their advantage.

## FOR TM
The stat boosts are disabled for now. We will see how the hijack attempts go for a bit. If the vast majority of the hijack attempts fail, the boosts with be added back,

## TODO

- [ ] Need to make / acquire a hud icon for hijack vampires and hijack changelings.

## Testing

Confirmed that the hijack vampires / changelings had the increased stats.

## Changelog
:cl:
tweak: Vampires and changelings can now roll hijack. ~~Those that do roll hijack will find the can use their powers for cheaper / more efficiently than if they were not a hijacker.~~
/:cl:

### CODE REVIEW

**Comment by Rurik123** [comment]
Reasoning on buffing cling/vamp if they roll hijack? Solely cause they lack high powered items like a chainsaw, or an emag?

**Comment by Rurik123** [comment]
While I follow your logic and I'm sympathetic to the tragic end that would befall a cling trying to storm shuttle with just an armblade, I've a few concerns to buffing Cling/Vamp during hijack. 

- We have to consider the pre-shuttle fights. Vampire/Cling will have full authorization to keep dead every one of its victims. Clings will be allowed to bait sec into kill zones and then hide corpses, vampires will be allowed to engage without needing blood as an excuse. Medbay can be targeted at will, hijacking clings/vamps will go harder with more "gamey" gear, no re-stealthing after objectives are done.
- If the above happened, an ERT would be sent to gun down the threat, but, correct me if I am wrong, ERT's are generally not sent over hijack objectives unless they are delaying a shuttle call. This means no manpower support for any victims kept dead. While this is the same for traitor, keep in mind traitor usually run out of anti-stuns to use while vamps only become more powerful, and clings plateau in strength so long as they pace themselves in combat. 
- Trifecta also has to be considered, and the utter weakness of evidence. Vamps and clings getting traitor gear already happens, but they are limited by their objectives. Here a changeling with chainsaw will have full clearance to kill everyone on the station _with_ chemical buffs to boot, _and_ no ERT's to back up security like what would normally happen if a cling went loud with traitor gear. 
- It should be noted everyone I know, and I imagine majority of the antag population always play harder when they roll hijack. Less stops, less "honor rules," less restraint on combo'ing gear pieces together, lot more murder, be it via toxin bombing or 357 onslaught. 
- Lastly, keep in mind just as with tot hijack, the antag is not alone. Even if nobody ahelped to officially help the hijacker, chaos compounds upon itself and one hijacker killing sec, and keeping them dead, all while thefts, murders, and blood draining is happening is going take the heat off the hijacker in many cases. I understand it is not always this way, but it is enough of time that its worth mentioning.

All those concerns aside, hemomancer being buffed on hijack is a very very bad idea. They already have every single tool they need to hijack. 8 man sec team armed with WTs will not stop them at shuttle even without a buff. If this PR rolls forward I'd highly recommend adding hemomancer class as an exception to the cost and cooldown reduction.

All that said, I'm not against hijack alone. Only the buffs atop of it. If I made false assumptions regarding ERT calling, please let me know as that will change things. Apologies for the post length.

**Comment by Kuzo1971** [comment]
I agree with every point Rurik made but to add on another point or two:

- RP. We already live in a culture where antag/sec RP is a constant complaint. I fail to see how increasing the chance to roll hijack, buffing said hijack, and bringing in a truly unlimited version of the antag, helps play into the whole MRP dynamic. I think we're often already at the limit for chaos personally. This PR views antag in a pocket without considering the entire picture, most of which is addressed in Rurik's reply. Now we're talking harder to detect tags lockering drained victims, clings absorbing, and having zero reasons to hold back in case. I fail to see how this is going to create a better experience for the WHOLE of the server rather than the one person that rolls it, and can go uninhibited. 
- My point two, you largely made yourself. "acquire high-powered weaponry to shred through crew, or the ever important emag for a quick launch." You already address antags crossing into tot gear that don't roll tot, and now no longer limited to even just that. Anything Sci provides, cargo provides, and botany provides is now entirely on the table.  I feel like everyone needs a reminder every now and then that cling tot/ cling hijack was removed for a reason. It's overpowered. Adding hijack to explore the limits isn't a justification for allowing people to break the round with no consequence when there are no consistent guidelines for ERT, no consistent guidelines on when a hijacker should call the shuttle, no consistent guidelines on escalation, these antags wouldn't be valid or stoppable by the crew(crew meaning more than sec, not even factoring how rolled sec would get because I'm a bit sec mained bias so let's keep it to the average crew), no answer to teaming on top of this gaming, and nothing stopping an antag from exploring everything in their touch. The server is nowhere near ready for something like this. 

I think this PR sounds fun on paper but would be chaotic in practice. I think we'd see combinations we didn't know existed, and now we have the average assistant, doctor, and scientist, getting rolled over by already base-strong antags with nothing stopping them to go loose. Given what people can already accomplish now without needing to roll hijack, I don't think this is a stove we need to touch before realizing we can get burned.  I can't agree with this idea on the logic that because it hasn't been done, it should be done. I don't see how it's good for more than one player. I think it underestimates how easy it is to go around the current downsides of these antags, and the reason that's not seen often is that it's not ALLOWED to in the status quo.

**Comment by Komrad822** [comment]
I personally love the idea of them having hijack. Changeling hijack I can see funny idea that normally you cant go with a tot hijack due to the fear of dieing. And I always felt that vampires upon reaching full power had to pull their punch, not allowing them this new full power to their full potential. I've gotten full power MULTIPLE times but I can only really use it on sec, everyone else its like "Oh hey its Eru, how are you? Would you like some coffee?" and RP wise its kinda eh. I dont have a reason to murder these people unless they attack me or I find them in maints and need to drain them. There is no fear of a vampire taking control of an entire station because players know they cant rule wise. 
The same with changlings, unless they REALLY need to, they cant just murder the entire department thats causing them headaches. They have to ahelp to kill the AI thats constantly bolting and shocking them since clings are not crew. Personally, Hijack is fun, lets you stretch your legs and sec can do the same against you after a certain degree.

Anyway, in terms of the buffs.... I feel a 33% COOLDOWN is a bit much. Garg needs it the most i'd say since their abilities are rather lack luster compared to others. I like the idea for the blood usage decrease and agree with Rurik that hemomancer needs its ownb numbers since claws give blood and its cooldowns are well balanced as is. Blood Bringer's Rites is really strong with their limiter removed.

Lastly While I do agree in terms of actually hijacking the shuttle, these antags without prep would be weak, Tot hijacks do not get anymore TC than other tots. So while above I mentioned which vamp should and shouldnt have these buffs, its kinda unfair to the tot hijacks who dont get anything more. ALSO, just have 3 command IDs to early launch the shuttle. 

TLDR: Love the idea, just think the buffs for vampire need more looking into.

**Comment by Kuzo1971** [comment]
> l. I've gotten full power MULTIPLE times but I can only really use it on sec, everyone else its like "Oh hey its Eru, how are you? Would you like some coffee?" and RP wise its kinda eh.



I'm sorry, this is just the perfect example of the disconnect going on with a PR like this. I'm genuinely confused as to what game we're aiming for when you're looking for an excuse to kill outside of sec, outside of draining. I find it odd that this can disappoint players on non-hijack and then in the same breath push for something that allows them to kill an entire branch for "being annoying." I'm not against hijack, but I get the impression this is the community shooting itself in the foot for trying to be a server that Paradise isn't. 

> The same with changelings, unless they REALLY need to, they cant just murder the entire department thats causing them headaches.


What are we really trying to do here, why are we looking to make this more common when the average player isn't even an antag or sec main. 

If something like this is going to go through, admins need to be way more lenient on validhunting if we're going to start moving in a more TG direction with thinking like this. Now that we could have more teams and more variety of round remove hijack combinations, the average player should be able to kill these antags way more than waiting on self-defense. All because players want to feel the power of taking control of the station. Believe it or not, if there's only 100 players online, only 20 are related to sec/antag. 

A checklist admins need before considering a pr like this

- Consistent ERT policy on hijacks
- Consistent policy on when a hijacker needs to actually call the shuttle
- Talks about what allows a branch, or player, to attack an antag, and what sec escalation may be like in response.

**Comment by silverplatedelta** [comment]
While I do see the appeal of allowing these antags to "use their full potential" I feel that this PR is extremely shortsighted. Currently, when traitor rolls hijack, it dooms a round to a very chaotic end. By design, traitors possess less innate ability for mass chaos than vampires and changelings and still manage to cause DEVISTATING damage to the station. Rolling traitor as a vampire or changeling will allow these players to cause the SAME level of devastation in addition to possessing strong near infinite use abilities, and in the case of changelings, near immortality.

Currently, vampire and changeling rounds allow a playerbase a reprieve from the chaos of hijackers, which in some cases brings the round to an end as soon as 30 minutes in. This change has the possibility of making mass bombings, plasma floods, delams, and crew genocide a nearly ever round occurrence. Its important to keep in mind that an overwhelming majority of our playerbase do not play security and do not play antagonist roles. Changes like this suggest a shift towards a more LRP environment more akin to a game of counterstrike than roleplay as crew on a space station.

Sure, hijacking is fun, but it often detracts from a round more than it adds to it. This simply punishes players who don't want to partake in PVP by making them violence outlets for antagonists. I strongly agree with Kuzo, in that the chaos on paradise is currently at its uppermost limit. Something that I feel HASNT been addressed in the above comments is the simple fact that an increase in hijacking will only serve to encourage rushing objectives earlier in the round. This will lead to more round removal earlier, which as I said previously will only serve to continue to alienate our players not interested in PVP. 

TLDR: Antagonists should make the round interesting, not end it. I would not be opposed to a full removal of the hijack objective in exchange for some more leeway depending on objectives. We need to stop alienating our non-PVP playerbase to allow someone who rolled a 1/50 chance to have fun.

**Comment by Qwertytoforty** [comment]
The boosts have been removed for now.

We will see how the TM goes.

If the clings or vampires get steamrolled constantly in hijack, their boosts will be returned to them.

**Comment by Kuzo1971** [comment]
> The boosts have been removed for now.
> 
> We will see how the TM goes.
> 
> If the clings or vampires get steamrolled constantly in hijack, their boosts will be returned to them.

I think you're going to find that the results of why this may not be the move are going to take months to realize when players sink into it more. A reminder that hijack isn't supposed to be easy, and I could see how a few bald antags rolling it, then red texting, could be used to justify this being full merged with the buffs.

**Comment by S34NW** [comment]
Please update this branch from master to resolve CI issues that were fixed in https://github.com/ParadiseSS13/Paradise/pull/21301

**Comment by matttheficus** [comment]
I think we have enough instances of chaos in the game at the moment (Xenos, hijackers, combat changelings that 1v6 security) along with the usual insanity of the round. I dont think adding more chaos in a time when we can barely get people to play as ERT/Security unless theyre a veteran due to the headache of dealing with uncontrolled chaos is good.

**Comment by OctusGit** [comment]
This a decision I dont think will improve the server in any bit. Like all the other said, this change is designed to appeal to antagonist players who make up a small margin of the community playerbase. This will force people to constantly be the victim of antagonist and force absolute gamer behavior (something that I assume even admins dislike greatly) that will just result in fast paced LRP gameplay. Paradise already struggles to keep its MRP tag with the amount of focus on antagonist and security, this wont help at all, it will only force the majority non pvp interested players to game or be gamed, that is a horrible outcome.

File: code/modules/antagonists/vampire/vampire_powers/vampire_powers.dm [hunk 1]
```diff
@@ -31,6 +31,11 @@
 /obj/effect/proc_holder/spell/vampire/self/create_new_targeting()
 	return new /datum/spell_targeting/self
 
+/obj/effect/proc_holder/spell/vampire/after_cast(list/targets, mob/user)
```
Comment (Miraviel) [inline code/modules/antagonists/vampire/vampire_powers/vampire_powers.dm:34]: This override causes Glare to use 2 charges instead of 1 per cast, and doesn't seem to affect anything else (since the 33% change was reverted)

**Review by Miraviel — CHANGES_REQUESTED** [review]

**Comment by Coolrune206** [comment]
Now that this has been testmerged for a bit, I can't say I'm the most fond of it. 
Vampires are at least the more interesting with their hijacks as their powers are somewhat suited for wide-area damage (hemo ult, umbrae ult, garg stomp, dantalion swarm) and as such don't inherently have to resort to typical boom tactics. If either of these two stay as hijackers, I'd probably say vampires would be the ones I'd favor.
Changelings are not interesting hijackers. They are just traitors without the emag to send the shuttle and instead resort to stationwide bombing or armblade + fleshmend + epi OD to try to hack through everyone they can. I feel like Changelings need more motivation to play their stealthy blend-in-and-replace people playstyle, and less of this motivation to be loud-Cling that murders everyone.

File: code/datums/spell_handler/vampire.dm [hunk 2]
```diff
@@ -40,7 +40,7 @@
 	vampire.bloodusable -= calculate_blood_cost(vampire)
 
 /datum/spell_handler/vampire/proc/calculate_blood_cost(datum/antagonist/vampire/vampire)
-	var/blood_cost_modifier = 1 + vampire.nullified / 100
+	var/blood_cost_modifier = (1 + vampire.nullified / 100)
 	var/blood_cost = round(required_blood * blood_cost_modifier)
 	return blood_cost
```
Comment (hal9000PR) [inline code/datums/spell_handler/vampire.dm:45]: whats with this change

File: code/modules/antagonists/changeling/datum_changeling.dm [hunk 3]
```diff
@@ -196,6 +201,19 @@
 		H.hud_used.lingchemdisplay.invisibility = 0
 		H.hud_used.lingchemdisplay.maptext = "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font face='Small Fonts' color='#dd66dd'>[round(chem_charges)]</font></div>"
 
+/datum/antagonist/changeling/hijackify() //Disabled initialy for TM. This will be re-added in if vampires / changelings constantly fall flat in hijack, and will be removed if they do perfectly fine.
+	. = ..()
+	//chem_storage = 100 //25% more
+	//chem_recharge_rate = 4 //25% more.
+	//genetic_points = 12 //20% more.
```
Comment (hal9000PR) [inline code/modules/antagonists/changeling/datum_changeling.dm:208]: commented out code
(and 100 chem storage isnt 25% more its 33% more)

File: code/modules/antagonists/traitor/datum_traitor.dm [hunk 4]
```diff
@@ -94,6 +94,7 @@
 	// Hijack objective.
 	if(prob(10) && !(locate(/datum/objective/hijack) in owner.get_all_objectives()))
 		add_objective(/datum/objective/hijack)
+		hijackify()
```
Comment (hal9000PR) [inline code/modules/antagonists/traitor/datum_traitor.dm:97]: I feel like `hijackify` should give the hijack objective

File: code/modules/antagonists/vampire/vamp_datum.dm [hunk 5]
```diff
@@ -25,6 +25,10 @@
 
 	/// list of the peoples UIDs that we have drained, and how much blood from each one
 	var/list/drained_humans = list()
+	/// How much blood will the vampire use on abilities? 200% = x2, 50% = x0.5
+	var/blood_efficency = 100
+	/// How long are vampire cooldowns compared to normal? 200 = x2, 50 = x0.5
+	var/cooldown_efficency = 100
```
Comment (hal9000PR) [inline code/modules/antagonists/vampire/vamp_datum.dm:31]: unused vars

File: code/modules/antagonists/vampire/vamp_datum.dm [hunk 6]
```diff
@@ -359,3 +375,8 @@
 /datum/hud/proc/remove_vampire_hud()
 	static_inventory -= vampire_blood_display
 	QDEL_NULL(vampire_blood_display)
+
+/datum/antagonist/vampire/hijackify() //I wanted to let a vampire chose 2 specializations here. However, we all know they would go hemo garg or hemo umbra and make everyone miserable. Instead, we are going to make vampires more efficent with hijack.
+	. = ..()  //Disabled initialy for TM. This will be re-added in if vampires / changelings constantly fall flat in hijack, and will be removed if they do perfectly fine.
+	//blood_efficency = 66 // 33% more efficent
+	//cooldown_efficency = 66 // 33% more efficent
```
Comment (hal9000PR) [inline code/modules/antagonists/vampire/vamp_datum.dm:382]: commented out stuff.
also, you can set vars to 200/3 to get an exact 66.66666

**Review by hal9000PR — CHANGES_REQUESTED** [review]
